src/main/java/uk/gov/hmcts/reform/sendletter/logging/AppInsights.java

```diff
@@ -25,40 +33,37 @@ public AppInsights(TelemetryClient telemetry) {
 
     // dependencies
 
-    public void trackServiceAuthentication(java.time.Duration duration, boolean success) {
-        telemetry.trackDependency(
-            AppDependency.AUTH_SERVICE,
-            AppDependencyCommand.AUTH_SERVICE_HEADER,
-            new Duration(duration.toMillis()),
-            success
-        );
-    }
-
-    public void trackFtpUpload(java.time.Duration duration, boolean success) {
-        telemetry.trackDependency(
-            AppDependency.FTP_CLIENT,
-            AppDependencyCommand.FTP_FILE_UPLOADED,
-            new Duration(duration.toMillis()),
-            success
-        );
-    }
-
-    public void trackFtpReportsDownload(java.time.Duration duration, boolean success) {
-        telemetry.trackDependency(
-            AppDependency.FTP_CLIENT,
-            AppDependencyCommand.FTP_DOWNLOAD_REPORTS,
-            new Duration(duration.toMillis()),
-            success
-        );
+    @Pointcut("@annotation(dependency)")
+    public void dependencyPointCut(Dependency dependency) {
+        // point cut definition
     }
 
-    public void trackFtpReportDelete(java.time.Duration duration, boolean success) {
-        telemetry.trackDependency(
-            AppDependency.FTP_CLIENT,
-            AppDependencyCommand.FTP_REPORT_DELETE,
-            new Duration(duration.toMillis()),
-            success
-        );
+    @Around("dependencyPointCut(dependency)")
+    public Object trackDependency(ProceedingJoinPoint joinPoint, Dependency dependency) throws Throwable {
+        Instant start = Instant.now();
+
+        try {
+            Object proceed = joinPoint.proceed();
```

Comment:
is this variable needed?

Reply:
yes. as can be expected in the java-logging too, think of a generalised version. this is a wrapper of any kind of method which is annotated with pointcut. method must return same thing whatever it returns